cocotb/regression.py

```diff
@@ -310,27 +369,21 @@ def _result_was():
             else:
                 self.log.error("Test error has lead to simulator shutting us "
                                "down", exc_info=exc_info)
-                self._add_failure(result)
-                self._store_test_result(test.module, test.funcname, False, sim_time_ns, real_time, ratio_time)
-                self.tear_down()
-                return
+                result_pass = False
+            sim_failed = True
 
         elif test.expect_error:
             if isinstance(result, test.expect_error):
                 self.log.info("Test errored as expected: " + _result_was())
             else:
-                self.log.info("Test errored with unexpected type: " + _result_was())
-                self._add_failure(result)
+                self.log.error("Test errored with unexpected type", exc_info=exc_info)
```

Comment:
Any reason for dropping `_result_was()` here?

Reply:
I think it's kind of redundant if you are printing the stacktrace, but I will add it back in, it might be used.